tests/integration-tests/images_factory.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import json
+import logging
+import re
+
+import yaml
+from utils import run_command
+
+
+class Image:
+    """Contain all static and dynamic data related to an image instance."""
+
+    def __init__(self, image_id, config_file, region):
+        self.image_id = image_id
+        self.config_file = config_file
+        self.region = region
+        with open(config_file) as conf_file:
+            self.config = yaml.safe_load(conf_file)
+        self.image_tags = None
+        self.creation_time = None
+        self.build_log = None
+        self.version = None
+        self.image_status = None
+        self.keep_logs = False
+
+    def build(self):
+        """Build image."""
+        command = ["pcluster", "build-image", "--id", self.image_id, "-r", self.region, "-c", self.config_file]
+        result = run_command(command).stdout
+        # TODO fix it to use json response
+        match = re.match(r"^{\'image\': \'(.*)\'}", result.split("\n")[6])
+        image_info = json.loads(match.group(1))
+        self._update_image_info(image_info)
+        return result
+
+    def delete(self, force=False):
+        """Delete image."""
+        command = ["pcluster", "delete-image", "--id", self.image_id, "-r", self.region]
+        if force:
+            command.append("--force")
+        if self.keep_logs:
+            command.append("--keep-logs")
+        result = run_command(command).stdout
+        return result
+
+    def describe(self):
+        """Describe image."""
+        logging.info("Describe image {0} in region {1}.".format(self.image_id, self.region))
+        command = ["pcluster", "describe-image", "--id", self.image_id, "-r", self.region]
+        result = run_command(command).stdout
+        # TODO fix it to use json response
+        match = re.match(r"^{\'image\': \'(.*)\'}", result.split("\n")[1])
+        image_info = json.loads(match.group(1))
+        self._update_image_info(image_info)
+        return result
+
+    def get_log(self):
+        """Get image build log."""
+        logging.info("Get image {0} build log.".format(self.image_id))
+        command = ["pcluster", "get-image-log-events", self.image_id, "-r", self.region]
```

Comment:
this command doesn't work, you have to pass the log-stream-name:
`get-image-log-events --log-stream-name 3.0.0/1 image-id`

Reply:
done